### PR TITLE
Support for AIX with xlC compiler

### PR DESCRIPTION
**Pull request by bb-nmushell**
These changes add support for the native xlC compiler of the AIX platform.

### CODE REVIEW

**Comment by saghul** [comment]
Since I have no idea about AIX we usually trust you folks actually using it, so LGTM :-) Maybe @mhdawson can take a quick peek though?

File: src/unix/atomic-ops.h [hunk 1]
```diff
@@ -33,6 +33,10 @@ UV_UNUSED(static int cmpxchgi(int* ptr, int oldval, int newval)) {
                         : "r" (newval), "0" (oldval)
                         : "memory");
   return out;
+#elif defined(_AIX) && (defined(__xlC__) || defined(__IBMC__) || defined(__IBMCPP__))
```
Comment (bnoordhuis) [inline src/unix/atomic-ops.h]: Just curious, doesn't `__xlC__` imply `__IBMC__` or `__IBMCPP__`?

**Comment by bnoordhuis** [comment]
LGTM but can you sign the commit with your real name?  You show up as `nmushell <nmushell@bloomberg.net>` now.

**Comment by bnoordhuis** [comment]
Oh, and can you make the first line of the first commit fit in 50 columns?  Cheers.

File: src/unix/atomic-ops.h (hunk 1 above)
Comment (bb-nmushell) [inline src/unix/atomic-ops.h]: http://beefchunk.com/documentation/lang/c/pre-defined-c/precomp.html
Looks like it is. I just used all of them. Though happy to make it just __xlC__.

Comment (bnoordhuis) [inline src/unix/atomic-ops.h]: I think that would be best / least confusing.

**Comment by mhdawson** [comment]
sorry been out at NodeConfEU and connectivity was pretty bad.  Looks ok to me other than the earlier comments about using only  __xlC__ and fixing up the header line.

**Comment by bb-nmushell** [comment]
Thanks for all the feedback!
- I have updated ```__xlC__``` macro in the code. 
- Updated the comment to be <50 characters
- Signed my commits

**Comment by saghul** [comment]
@bb-nmushell :+1: merging

**Comment by saghul** [comment]
Cheers, Neil. Landed in cf0e5b4db2f2b2651300d1598f14690f87762f10 and 029a43997d60c3a3d2ec15fdc12cfad8b949cbf7

**Pull request closed without merging**